extensions/wikia/AdEngine/js/MessageListener.js

```diff
@@ -55,7 +55,9 @@ define('ext.wikia.adEngine.messageListener', [
 					log(['Event matching', msg, callback], 'debug', logGroup);
 
 					callback.fn(JSON.parse(msg.data).AdEngine);
-					callbacks.splice(i, 1);
+					if (!callback.match.infinite) {
```

Comment:
_Blocks are nested too deeply. (4) [W073]_

Consider moving few `if`s to a separate function.